projects/js-packages/components/components/record-meter-donut/index.tsx

```diff
@@ -0,0 +1,104 @@
+import { __ } from '@wordpress/i18n';
+import React from 'react';
+import './style.scss';
+export type DonutMeterProps = {
+	/**
+	 * Total number of items for the donut meter.
+	 */
+	totalCount: number;
+	/**
+	 * Count for the given item
+	 */
+	segmentCount: number;
+	/**
+	 * Color code for the background color for the item.
+	 * If not provided, defaults to Jetpack Green
+	 */
+	backgroundColor?: string;
+
+	/**
+	 * thickness for the chart border
+	 * If not provided, defaults to 3.5
+	 */
+	thickness?: string;
+	/**
+	 * width for the full chart size
+	 * If not provided, defaults to 64px
+	 */
+	donutWidth?: string;
+};
+
+/**
+ * Generate record meter donut bar
+ *
+ * @param {DonutMeterProps} props - Props
+ * @returns {React.ReactElement} - JSX element
+ */
+const DonutMeter: React.FC< DonutMeterProps > = ( {
+	totalCount,
+	segmentCount,
+	backgroundColor = '#00BA37', // jetpack green fallback
+	thickness = '3.5',
+	donutWidth = '64px',
+} ) => {
+	const count = () => {
+		// get count as a percent value
+		return ( segmentCount / totalCount ) * 100;
+	};
+
+	return (
+		<div className="donut-meter" aria-hidden="true">
```

Comment:
Instead of hiding the chart from screen readers, maybe we can mark it up a bit more?

- Wrap the DIV in a FIGURE element.
- Add ARIA-LABELLEDBY element.
- Add TITLE & DESCRIPTION elements.
- Make the TITLE & DESCRIPTION values props.
- Set the role attribute to IMG.

```
<figure>
  <div class="figure-content">
    <svg width="100%" height="100%" viewBox="0 0 42 42" class="donut" aria-labelledby="beers-title beers-desc" role="img">
      <title id="beers-title">Beers in My Cellar</title>
      <desc id="beers-desc">Donut chart showing 10 total beers. Two beers are Imperial India Pale Ales, four beers are Belgian Quadrupels, and three are Russian Imperial Stouts. The last remaining beer is unlabeled.</desc>
```

This might eliminate the need for the summary table following the meter as well.

Reply:
I like this idea a lot. Noting that this table is almost identical to the RecordMeterBar component chart which serves the same purpose. So it might be worth dropping into that component and making the same changes 😄